### ISSUE TITLE
missing_panics_doc triggers on `debug_assert`

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#6739]
Lint name:

I tried this code:

```rust
#[inline]
fn temper(mut x: u32) -> u32 {
    x ^= x >> 11;
    x ^= (x << 7) & 0x9d2c_5680;
    x ^= (x << 15) & 0xefc6_0000;
    x ^= x >> 18;
    x
}

impl Mt19937GenRand32 {
    pub fn next_u32(&mut self) -> u32 {
        // Failing this check indicates that, somehow, the structure
        // was not initialized.
        debug_assert!(self.idx != 0);
        if self.idx >= N {
            fill_next_state(self);
        }
        let Wrapping(x) = self.state[self.idx];
        self.idx += 1;
        temper(x)
    }
}
```

I expected to see this happen: no lint

Instead, this happened:

```
error: docs for function which may panic missing `# Panics` section
   --> src/mt.rs:271:5
    |
271 | /     pub fn next_u32(&mut self) -> u32 {
272 | |         // Failing this check indicates that, somehow, the structure
273 | |         // was not initialized.
274 | |         debug_assert!(self.idx != 0);
...   |
280 | |         temper(x)
281 | |     }
    | |_____^
    |
note: the lint level is defined here
   --> src/lib.rs:13:9
    |
13  | #![deny(clippy::pedantic)]
    |         ^^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::missing_panics_doc)]` implied by `#[deny(clippy::pedantic)]`
note: first possible panic found here
   --> src/mt.rs:274:9
    |
274 |         debug_assert!(self.idx != 0);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_panics_doc
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Meta

- `cargo clippy -V`: `clippy 0.1.52 (e9920ef 2021-02-11)`
- `rustc -Vv`:
  ```
  rustc 1.52.0-nightly (e9920ef77 2021-02-11)
  binary: rustc
  commit-hash: e9920ef7749d11fc71cc32ca4ba055bcfeaab945
  commit-date: 2021-02-11
  host: x86_64-unknown-linux-gnu
  release: 1.52.0-nightly
  LLVM version: 11.0.1
  ```

**Comment by lopopolo** [comment]
oops this comes from the indexing operation. sorry for the noise.
